tests/ops/service/connectors/fides/test_fides_client.py

```diff
@@ -0,0 +1,427 @@
+        )
+        assert request.body == "key1=value1&key2=value2"
+
+        # test form data passed as dict
+        request = test_fides_client.authenticated_request(
+            "POST",
+            path="/testpath",
+            query_params={"param1": "value1", "param2": "value2"},
+            data={"key1": "value1", "key2": "value2"},
+        )
+        assert "Authorization" in request.headers
+        assert request.headers["Authorization"] == f"Bearer {SAMPLE_TOKEN}"
+        assert request.method == "POST"
+        assert (
+            request.url
+            == test_fides_client.uri + "/testpath?param1=value1&param2=value2"
+        )
+        assert request.body == "key1=value1&key2=value2"
+
+        # test body passed as string literal
+        request = test_fides_client.authenticated_request(
+            "POST",
+            path="/testpath",
+            query_params={"param1": "value1", "param2": "value2"},
+            data="testbody",
+        )
+        assert "Authorization" in request.headers
+        assert request.headers["Authorization"] == f"Bearer {SAMPLE_TOKEN}"
+        assert request.method == "POST"
+        assert (
+            request.url
+            == test_fides_client.uri + "/testpath?param1=value1&param2=value2"
+        )
+        assert request.body == "testbody"
+
+        # test json body passed as a dict
+        request = test_fides_client.authenticated_request(
+            "POST",
+            path="/testpath",
+            query_params={"param1": "value1", "param2": "value2"},
+            json={"field1": "value1"},
+        )
+        assert "Authorization" in request.headers
+        assert request.headers["Authorization"] == f"Bearer {SAMPLE_TOKEN}"
+        assert request.method == "POST"
+        assert (
+            request.url
+            == test_fides_client.uri + "/testpath?param1=value1&param2=value2"
+        )
+        assert request.body == b'{"field1": "value1"}'
+
+        # test json body passed as a list
+        request = test_fides_client.authenticated_request(
+            "POST",
+            path="/testpath",
+            query_params={"param1": "value1", "param2": "value2"},
+            json=[{"field1": "value1"}],
+        )
+        assert "Authorization" in request.headers
+        assert request.headers["Authorization"] == f"Bearer {SAMPLE_TOKEN}"
+        assert request.method == "POST"
+        assert (
+            request.url
+            == test_fides_client.uri + "/testpath?param1=value1&param2=value2"
+        )
+        assert request.body == b'[{"field1": "value1"}]'
+
+    # The below polling tests need to rely on response mocking, since we can't control
+    # the status of a given privacy request on a remote server.
+    # As such, they make some assumptions about the status API and are susceptible to
+    # false positives if the status API changes, and we don't adjust our tests here!
+
```

Comment:
i think the approach in https://github.com/ethyca/fides/pull/1860 generally may be better than what i've got here, i didn't like the mocking. that being said, the mocking allows us to test the actual blocking functionality in `test_poll_for_completion_blocks_till_complete` - i don't _think_ we can really do that without mocking.